yolo_to_yolo/data_transformers.py

```diff
@@ -47,3 +47,137 @@ def __call__(self, input_data: YoloImageData) -> Iterable[YoloImageData]:
     def include(classname: str) -> bool:
         return classname != 'background' and len(classname) > 1
 
+class BBoxToCropTransformer(YoloDataTransformer):
+    """
+    Converts bounding box labels to cropped images
+
+    Args:
+    min_size: Minimum size of the cropped image (width, height)
+    min_padding: Padding to add around the bounding box
+    min_char_overlap: Minimum overlap between shape and character bounding boxes to consider them a match (0 to 1)
+        (This is used to filter out char bboxes that don't overlap enough with the shape bboxes. Default is 0, meaning it just has to overlap a little bit to be considered a match)
+    """
+    def __init__(self, min_size: tuple[int, int] = (0, 0), min_padding: int = 0, min_char_overlap: float = 0):
+        self.min_size = min_size
+        self.min_padding = min_padding
+        self.min_char_overlap = min_char_overlap
+
+    def __call__(self, input_data: YoloImageData) -> Iterable[YoloImageData]:
+        img_height, img_width = input_data.image.shape[:2]
+        
+        # I hope this is right
+        shape_labels = [label for label in input_data.labels if "shape" in label.classname]
+        char_labels = [label for label in input_data.labels if "character" in label.classname]
```

Comment:
They're actually not this. There's a BACKGROUND/background label, then all one character classes are characters, I don't know how to distinguish between color and shapes.

So, you'll need to write some filters. I recommend you add it as a property function of the YoloLabel class for reusability and have it return an enum of `ClassType`. I don't know what a good way to oo that will be.

Here are the [competition rules](https://static1.squarespace.com/static/5d554e14aaa5e300011a4844/t/65037985fd52061081866123/1694726534579/suas-2023-rules.pdf). The target specifications are at the bottom of the 19th page.

Next time, try not to make assumptions because things usually aren't what they seem. Just ask instead of hoping.